bvm/ballerina-core/src/main/java/org/ballerinalang/bre/bvm/BVM.java

```diff
@@ -5399,11 +5356,11 @@ public FreezeStatus(State state) {
             this.currentState = state;
         }
 
-        private void setFrozen() {
+        public void setFrozen() {
```

Comment:
Is there a possibility of moving `FreezeStatus` somewhere else and leaving these methods private, or will there be a circular dependency? 

Since this is now public, the freeze status can be modified via the BValue right? This was left private to only allow modification via the `.freeze()` call itself.

Reply:
In order to keep it private, we need to move FreezeStatus to utils stdlib. Then we need to add a dependency from core to stdlib, which will introduce a circular dependency. Hence, it's not possible.